### ISSUE TITLE
Could autocomplete be used to meet the proposed redundant entry SC auto-populate clause?

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#1194]
If a developer used the autocomplete property in HTML on a field would that be sufficient to meet the auto-populate clause of the newly proposed Redundant entry success criteria?

**Comment by alastc** [comment]
My initial reaction is no, because the content isn't doing the filling in, but it's a fine point.

Auto-population is reliable if the site does it, but if the user doesn't have it on (e.g. for a shared computer) then it won't work.

**Comment by patrickhlauke** [comment]
i'd agree with @alastc's take here

**Comment by guyhickling** [comment]
Autocomplete only applies to certain specific items of personal data, it doesn't apply to most data. This SC covers any data item so autocomplete would not work for most of that. 

But for personal data that is covered by autocomplete, it seems to me that W3C should state, perhaps in the Understanding, whether or not autocomplete should be a permitted way of complying with this SC. Developers, as well as testers and auditors, will need to know what's allowed.

**Comment by johnfoliot** [comment]
As Alastair and Patrick notes, "auto-completion" is dependent on browser
settings and  thus not always dependable.

But perhaps tagging inputs with either @autocomplete or any or the emergent
Personalization attributes would be sufficient, as the author has supplied
the intent programmatically.

JF

(Sent from my mobile, apologies for any spelling mistakes)

**Comment by alastc** [comment]
> But perhaps tagging inputs with either @autocomplete or any or the emergent
Personalization attributes would be sufficient, as the author has supplied
the intent programmatically

Thinking about it more, I'm inclined to keep some separation between this and autocomplete, as this wasn't intended to be filled in by the browser, but something for the site to action (filled in by the site).

I would suggest a line in the understanding saying that autocomplete does **not** fulfill this criterion would suffice, if agreed?

**Comment by Helixopp** [comment]
We specifically discussed autocomplete when agreeing on techniques. We decided no, autocomplete doesn't pass because it requires users to rely on their memory, more specifically recall, as opposed to recognition. That is a specific cognitive accessibility barrier that this SC was meant to address.

**Comment by alastc** [comment]
Noting that Michael Gower proposed a different update:
<hr>
Requiring people to recall information already entered in the previous steps can cause them to give up or enter incorrect information. The autocomplete feature of browsers is not considered a sufficient technique for several reasons:
1) it requires the user to initiate a response before it provides autocomplete suggestions
2) it is reliant on historically entered data, which may not be relevant or valid for the current user or context
3) the values available with the autocomplete attribute may not address data requirements.
<hr>
If we get more questions about this aspect, we can use it.

**Comment by jake-abma** [comment]
the longer I think about the reasons the more I'm not so sure:

--------------------------

> 1. it requires the user to initiate a response before it provides autocomplete suggestions

the "information previously entered by (or provided to) the user" is exactly that (and a content author JS solution might be the same as the UA solution)

> 2. it is reliant on historically entered data, which may not be relevant or valid for the current user or context

Again, the SC text is about "information previously entered by (or provided to) the user" which is "historically entered data" even if it was just a minute ago. This will be relevant and available with autocomplete although you might also have more options to choose from NOT relevant.

> 3. the values available with the autocomplete attribute may not address data requirements.

Ok, but IS it available for the user to select? If so, it passes the text.

--------------------------

I think we're trying to find reasons for excluding autocomplete from a wrong perspective, also because an author might be simulating / copying the same approach as a UA does.

**My best shot at this would be to mention that we "only allow a mechanism provided by the content creator" and NOT "a mechanism (or use of a mechanism) provided by a UA like the @autocomplete functionality"** (thus making it clear it's about the sender and the receiver, not the UA in between)

This way we do not need to find reasons for excluding a UA functionality while the reasons are not always very solid.

**Comment by alastc** [comment]
Hi Jake,

The change that has been incorporated into the understanding doc is:
> The autocomplete feature of browsers is not considered sufficient because it is the content (the web site) that needs to provide the stored information for a redundant entry, or avoid asking for the same information again.

That's pretty similar to what you said above?

**Comment by jake-abma** [comment]
Yep, clear and thanks!
